Yehor/reactNetflix/src/components/search-panel/SearchPanel.tsx

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import "./SearchPanel.css";
+import Button from "../button";
+import { parse } from "query-string";
+import * as History from "history";
+
+interface ISearchPanelState {
+  value: string;
+  searchBy: SearchBy;
+}
+interface IhandleSearchChangeProps {
+  handleSearchChange({
+    search,
+    searchBy,
+  }: {
+    search: string;
+    searchBy: string;
+  }): void;
+  location: History.Location;
+}
+enum SearchBy {
+  Title = "title",
+  Genre = "genre",
+}
+
+class SearchPanel extends React.Component<IhandleSearchChangeProps, ISearchPanelState> {
+  state: ISearchPanelState = {
+    value: "",
+    searchBy: SearchBy.Title,
+  };
+
+  searchByTitle = () => {
+    if (this.state.searchBy === SearchBy.Genre) {
+      this.setState({
+        value: "",
+        searchBy: SearchBy.Title,
+      });
+    }
+  };
+  searchByGenre = () => {
+    if (this.state.searchBy === SearchBy.Title) {
+      this.setState({
+        value: "",
+        searchBy: SearchBy.Genre,
+      });
+    }
+  };
+  handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      value: e.target.value,
+    });
+  };
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.props.handleSearchChange({
+      search: this.state.value,
+      searchBy: this.state.searchBy,
+    });
+  };
+
+  componentDidMount() {
+    const querySrch = parse(this.props.location.search) as {
+      searchBy: string;
+      search: string;
+    };
+    const { searchBy, search } = querySrch;
+    // verify searchBy
+    if (searchBy) {
```

Comment:
if you rewrite this statements to the code below, nothing will change, but there will be less code. 
`if (searchBy && searchBy === SearchBy.Genre) {
    this.setState({ searchBy: SearchBy.Genre }) 
} else {
     this.setState({ searchBy: SearchBy.Title }) 
};
`
 }`